### ISSUE TITLE
Updated code for the new paper published in June 2021?

### ISSUE THREAD

**Issue by harrishsmith** [farjon/Leaf-Counting#2]
Hi Guy,

I wanted to make sure if this is the updated code repository for the new paper: Farjon, G., Itzhaky, Y., Khoroshevsky, F., & Bar-Hillel, A. (2021). Leaf Counting: Fusing Network Components for Improved Accuracy. Frontiers in Plant Science, 12, 1063. OR the previous paper: Itzhaky, Y., Farjon, G., Khoroshevsky, F., Shpigler, A., & Bar-Hillel, A. (2018, September). Leaf counting: Multiple scale regression and detection using deep CNNs. In BMVC (p. 328).

**Comment by farjon** [comment]
Hi Harrish
This repo includes the updated code for the new paper from 2021
Both of the training scripts are located at the 'bin' directory
'train_det' is our DRN pipe
'train_reg' is our MSR pipe

**Comment by harrishsmith** [comment]
Thank you much.

**Comment by harrishsmith** [comment]
Hi Guy,

Thank you for the response. I am trying to run the code. But unfortunately, I am getting the 'No module named 'GetEnvVar'' error here: https://github.com/farjon/Leaf-Counting/blob/797869666e475faf59682913dcd1c3d11a0f205d/LeafCounting/bin/train_reg.py#L30 . Can you guide how the training environment needs to be maintained? I installed all dependencies in requirements.txt

**Comment by farjon** [comment]
GetEnvVar function is used to point the system to where the data, pretrained models, and any other large data files
You can either replace it with your own function (which operates as an hub - if you request the ‘models’ directory path, you write GetEnvVar(‘ModelsPath’) and receive the path to your pretrained models directory), or you can just replace it with a specific path (please note that we used GetEnvVar function in several places)
For example, see line 432 in the script ‘train_reg.py’

args.data_path = os.path.join(GetEnvVar('DatasetsPath'), 'Phenotyping Datasets', 'Plant phenotyping', 'data_2', 'CVPPP2017_LCC_training', 'training', '{}'.format(args.dataset))

**Comment by harrishsmith** [comment]
Thank you. I am trying to replicate the results. Any suggestions to replicate the results? 

1. I am assuming 'reg_fpn_p3_p7_mle' is the correct option for counting without detection. Is it correct?
2. You mentioned that "To predict the number of leaves open LeafCounting\bin\predict_number_of_leaves.py". Can I use "evaluate_counting.py" for result evaluation for both countings without detection and countings with detection? Is there any distinction between these two scripts?

**Comment by farjon** [comment]
1. correct
2. "evaluate_counting.py" accepts the real number of leaves as input, and is used to evaluate the model. It can be used for both models (with/without detection)
To simply predict the number of leaves (without any labeled data) you can run "predict_number_of_leaves.py"

**Comment by harrishsmith** [comment]
Thank you much.
